src/Core/Services/StateService.cs

```diff
@@ -1724,15 +1724,15 @@ public async Task SetAvatarColorAsync(string value, string userId = null)
         {
             var reconciledOptions = ReconcileOptions(new StorageOptions { UserId = userId },
                 await GetDefaultStorageOptionsAsync());
-            var key = Constants.AvatarColorKey;
+            var key = Constants.AvatarColorKey(userId);
```

Comment:
`userId` should be `reconciledOptions.UserId` in order to support null fallback to the current active user